cmake/Platform/Common/Install_common.cmake

```diff
@@ -196,7 +201,17 @@ function(ly_setup_target OUTPUT_CONFIGURED_TARGET ALIAS_TARGET_NAME absolute_tar
         string(GENEX_STRIP ${include} include_genex_expr)
         if(include_genex_expr STREQUAL include) # only for cases where there are no generation expressions
             # Make the include path relative to the source dir where the target will be declared
-            cmake_path(RELATIVE_PATH include BASE_DIRECTORY ${absolute_target_source_dir} OUTPUT_VARIABLE target_include)
+            cmake_path(IS_PREFIX CMAKE_BINARY_DIR ${include} NORMALIZE include_directory_child_of_build)
+            if(include_directory_child_of_build)
+                # Some autogen files are placed in the build folder so remove the build folder prefix
+                # and use it to calculate the relative path  
+                cmake_path(RELATIVE_PATH include BASE_DIRECTORY ${CMAKE_BINARY_DIR} OUTPUT_VARIABLE rel_include)
+                cmake_path(SET base_path ${LY_ROOT_FOLDER})
+                cmake_path(APPEND base_path ${rel_include} OUTPUT_VARIABLE absolute_include)
+                cmake_path(RELATIVE_PATH absolute_include BASE_DIRECTORY ${absolute_target_source_dir} OUTPUT_VARIABLE target_include)
+            else()
+                cmake_path(RELATIVE_PATH include BASE_DIRECTORY ${absolute_target_source_dir} OUTPUT_VARIABLE target_include)
+            endif()
```

Comment:
Well this logic is a bit of fuzzy in that it would only work for Gems that are part of the engine.

This would form a path of `<source-engine>/External/<gem-name>-<sha1-hash>/Code/Generated/<target-name>` which doesn't actually exist as a folder.

It would then create a relative path from the target source directory which would be something such as `<source-engine>/Gems/<gem-name>/Code`.

Making the path of `<source-engine>/External/<gem-name>-<sha1-hash>/Code/Generated/<target-name>`  relative to `<source-engine>/Gems/<gem-name>/Code` would transform the path to go up 3 directories from the Gem folder before diving into CodeGen path.
`../../../External/<gem-name>-<sha1-hash>/Code/Generated`.

So that would work as the Gems being built into the SDK resides under the engine root. 
If not the paths would be off. 
For example if a development wanted to make their own SDK with external Gems registered to the `engine.json` that resides outside of the <engine-root>, then the relative path formed becomes dependent on files outside of the <engine-root>.

That being said O3DE can just state as a requirement that when making an SDK, all Gems need to be contained within the engine root and registered with the `engine.json`.

Reply:
This is a good call out and I created a GHI to follow up on how it would fail for gems outside the engine that are included in an install https://github.com/o3de/o3de/issues/17241